### PR TITLE
[remove:2] Switch to promises and Add remove function

### PR DESCRIPTION
**Pull request by d2lam**
- Switch to promises
- Add remove. This resolves to null if success, rejects if fails.  
- Update README

Depends on this https://github.com/screwdriver-cd/datastore-base/pull/14

Related to https://github.com/screwdriver-cd/models/issues/105

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -112,26 +142,24 @@ class Dynamodb extends Datastore {
...
         const updateOptions = {
             expected: { id }
         };
 
-        if (!client) {
-            return callback(null, null);
-        }
+        return new Promise((resolve, reject) => {
+            if (!client) {
+                return resolve(null);
```
Comment (stjohnjohnson) [inline index.js]: Hmm, why does this one resolve instead of reject for bad table name?

**Review by stjohnjohnson — CHANGES_REQUESTED** [review]
Oh boy.  I love that it's all promised (kinda just wrapped up right now).  Just update the update failure.

File: index.js (hunk 1 above)
Comment (d2lam) [reply index.js]: That's true. I'm not sure why. This is just translating from the old code. I feel like it should rejects here too... Maybe it was a bug.

Comment (d2lam) [reply index.js]: @stjohnjohnson  Fixed

**Review by stjohnjohnson — APPROVED** [review]
Looks good! 👍

File: test/index.test.js [hunk 2]
```diff
@@ -163,61 +164,58 @@ describe('index test', () => {
...
 
-        it('fails when given an unknown table name', (done) => {
+        it('fails when given an unknown table name', () =>
             datastore._get({
                 table: 'tableUnicorn',
                 params: {
                     id: 'doesNotMatter'
                 }
-            }, (err, data) => {
+            }).then(() => {
```
Comment (minz1027) [inline test/index.test.js]: Why not .catch((err) => {}) here? I just feel throw new Error('Oops') in the then is a bit confusing.

File: test/index.test.js [hunk 3]
```diff
@@ -163,61 +164,58 @@ describe('index test', () => {
...
+                throw new Error('Oops');
+            }, (err) => {
```
Comment (minzcmu) [inline test/index.test.js]: Why not `.catch((err) => {})` here? I just feel `throw new Error('Oops')` in the `then`is a bit confusing.

Comment (d2lam) [reply test/index.test.js]: I changed the `(err)` inside `then` to `.catch(err)` to be less confusing. 
The `throw new Error('Oops')` inside the `then` is necessary to ensure that it should not get there.

Comment (minzcmu) [reply test/index.test.js]: Cool! 👍

**Review by minzcmu — APPROVED** [review]
winkwink

**Pull request merged**